### ISSUE TITLE
real-time continuous webcam prediction

### ISSUE THREAD

**Issue by hanako94** [harvitronix/five-video-classification-methods#75]
hi, may i know, instead of reading an original video file, extract features and sequences, and make prediction,
is that possible to let the program read from webcam and predict in real-time continuously  with different set of actions? for example, apply-lipstick, followed by apply eye make up.

**Comment by mesmerx** [comment]
you can grab n frames, put in a tensor, or make a temp file, predict this file, and then remove/restart the loop this way you predict in real time, i think

**Comment by hanako94** [comment]
is it possible to reach real time? because we not only need to create frames while streaming, we need to extract the features and create the sequence.npy file for prediction.
i could take quite sometime to extract sequence of features.

**Comment by mesmerx** [comment]
Train the model , and  then with a loaded model  uses opencv  extract lhe first 300 frames and  save  in  array, predict  that array, and  in  next  loop  add  one  frame  and remove one, so  you  can  predict  in  real time

**Comment by mesmerx** [comment]
I believe  this   could run at 20 fps  or more

**Comment by hanako94** [comment]
I’m using lstm model. 
Yes, I used the opencv to extract frames. But then it takes some time to extract the features from the frames created(as the author mentioned, lstm will use the feature extracted for prediction).  
I think the problem is the feature extraction part. 
Ps: I’m very new to tf and machine learning. Thanks for explaining to me so patiently.

**Comment by mesmerx** [comment]
when  i finish  my  actual  model i will ttry  make  that,if you can wait

**Comment by hanako94** [comment]
It’s good to know you will try to make this. I appreciate that so much. 
Hope you can finish ur model soon. Haha

**Comment by hanako94** [comment]
another question to ask, is that possible to deploy this keras trained model (hdf5) to android ?

**Comment by mesmerx** [comment]
I cant se why  not, Just a tip, Save  only weights  and  then  save  json. Its  faster oi load

**Comment by mesmerx** [comment]
I made  a custom  demo  script, i will push to my git

**Comment by hanako94** [comment]
great!! i will go to have a try

**Comment by hanako94** [comment]
hi, i couldn't find your custom demo script

**Comment by hanako94** [comment]
@mesmerx  i did not see the demo script that you mentioned.

**Comment by mesmerx** [comment]
I have not  pushed yet, Just making some work

**Comment by hanako94** [comment]
@mesmerx  have you done it? i still unable to solve the problem

**Comment by Lelo123** [comment]
@mesmerx and @hanako94

**Comment by hanako94** [comment]
you can convert the keras .hdf5 to tensorflow .pb format and then plug to your tensorflow android project
https://github.com/amir-abdi/keras_to_tensorflow

**Comment by Lelo123** [comment]
Thanks. Justiça one more questionário, i'm Stiller very new to machine learning, but wasn't supose to exist a Validation and test set? I'm this cases the Validation set seems to be the test set for me

**Comment by Lelo123** [comment]
@hanako94

**Comment by drewgash** [comment]
@mesmerx 

> when i finish my actual model i will ttry make that,if you can wait

Were you able to complete this?
